lib/ansible/modules/net_tools/speedtest.py

```diff
@@ -0,0 +1,218 @@
+                    'supported_by': 'community'}
+
+RETURN = '''
+bytes_received:
+    description: bytes recevied during download testing
+    returned: always
+    type: int
+    sample: 315786858
+bytes_sent:
+    description: bytes sent during upload testing
+    returned: always
+    type: int
+    sample: 105914368
+client:
+    description: information about local client
+    returned: always
+    type: dict
+download:
+    description: download speed in bytes per second
+    returned: always
+    type: float
+    sample: 252432440.16133374
+ping:
+    description: latency to chosen server
+    returned: always
+    type: float
+    sample: 5.491
+server:
+    description: information about the remote test server
+    returned: always
+    type: dict
+share:
+    description: URL to speedtest.net share image, will be null if share=False
+    returned: always
+    type: str
+    sample: http://www.speedtest.net/result/1234.png
+timestamp:
+    description: ISO8601 timestamp recorded at the start of the test
+    returned: always
+    type: str
+    sample: "2018-04-02T16:00:20.036880Z"
+upload:
+    description: upload speed in bytes per second
+    returned: always
+    type: float
+    sample: 81690942.203669
+'''
+
+import traceback
+
+try:
+    import speedtest
+    HAS_SPEEDTEST = True
+except ImportError:
+    HAS_SPEEDTEST = False
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def run_speedtest(module):
+    s = speedtest.Speedtest(
+        source_address=module.params['source'],
+        secure=module.params['validate_certs'],
+        timeout=int(module.params['timeout']),
+    )
+
+    if module.params['mini']:
+        s.get_best_server(s.set_mini_server(module.params['mini']))
+    else:
+        s.get_servers(
+            module.params['servers'],
+            module.params['exclude']
+        )
+        s.get_best_server()
+
+    if module.params['download']:
+        s.download()
+    if module.params['upload']:
+        s.upload(pre_allocate=module.params['pre_allocate'])
+    if module.params['share']:
+        s.share()
+
+    module.exit_json(**s.results.dict())
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            servers=dict(required=False, type='list', elements='int', default=[]),
+            exclude=dict(required=False, type='list', elements='int', default=[]),
+            mini=dict(required=False, type='str'),
+            download=dict(type='bool', default=True),
+            upload=dict(type='bool', default=True),
+            share=dict(type='bool', default=False),
+            source=dict(required=False, type='str'),
+            validate_certs=dict(type='bool', default=True),
+            timeout=dict(type='int', default=10),
+            pre_allocate=dict(type='bool', default=True),
+        ),
+        mutually_exclusive=[
+            ['servers', 'mini'],
+        ],
+        required_one_of=[
+            ['download', 'upload'],
```

Comment:
Why is one of `download` and `upload` required, given that they both default to true?